ngraph/core/src/op/color_convert_nv12.cpp

```diff
@@ -0,0 +1,202 @@
+                                               ColorConversion format)
+    : Op({arg_y, arg_uv}),
+      m_format(format),
+      m_single_tensor(false) {
+    constructor_validate_and_infer_types();
+}
+
+void ov::op::v8::ConvertColorNV12::validate_and_infer_types() {
+    NGRAPH_OP_SCOPE(v8_Convert_NV12_validate_and_infer_types);
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(0) == element::u8 || get_input_element_type(0) == element::f32,
+                          "Y input shall have u8 or f32 precision, got ",
+                          get_input_element_type(0));
+    const auto& shape_y = get_input_partial_shape(0);
+    NODE_VALIDATION_CHECK(this,
+                          shape_y.rank().is_static() && shape_y.rank().get_length() == 4,
+                          "Y input shall not have fully dynamic shape and have 'NCHW' layout");
+
+    NODE_VALIDATION_CHECK(this,
+                          shape_y[C_DIM].is_dynamic() || shape_y[C_DIM].get_length() == 1,
+                          "Y channels dimension shall be either dynamic or equal to 1. Current value is ",
+                          shape_y[C_DIM].get_length());
+
+    auto out_shape = shape_y;
+    out_shape[C_DIM] = 3;  // 3 is number of channels (R, G, B)
+    if (m_single_tensor) {
+        if (shape_y[H_DIM].is_static()) {
+            // E.g. if input shape height is 720 for NV12, then real image height is 720 * 2 / 3 = 480
+            out_shape[H_DIM] = shape_y[H_DIM].get_length() * 2 / 3;
+        }
+    } else {
+        NODE_VALIDATION_CHECK(this,
+                              get_input_element_type(1) == get_input_element_type(0),
+                              "UV input ",
+                              get_input_element_type(1),
+                              " shall have same precision as Y input ",
+                              get_input_element_type(0));
+        const auto& shape_uv = get_input_partial_shape(1);
+        NODE_VALIDATION_CHECK(this,
+                              shape_uv.rank().is_static() && shape_uv.rank().get_length() == 4,
+                              "UV input shall not have fully dynamic shape and have 'NCHW' layout");
+        // Verify that height for Y input is 2 times bigger than input height for UV
+        if (shape_y[H_DIM].is_static() && shape_uv[H_DIM].is_static()) {
+            NODE_VALIDATION_CHECK(this,
+                                  shape_y[H_DIM].get_length() == shape_uv[H_DIM].get_length() * 2,
+                                  "Y input height shall be 2 times bigger that UV input height: Y height = ",
+                                  shape_y[H_DIM].get_length(),
+                                  " UV height = ",
+                                  shape_uv[H_DIM].get_length());
+        }
+        // Verify that width for Y input is 2 times bigger than input width for UV
+        if (shape_y[W_DIM].is_static() && shape_uv[W_DIM].is_static()) {
```

Comment:
I believe we should be either both static or both static.

Cases like dynamic + static are not allowed, am I right?

Reply:
Fixed in merged PR #7601